lib/app/AppState.js

```diff
@@ -0,0 +1,107 @@
+/**
```

Comment:
We need to separate state model and state repository. Let's use leveldb as storage instead of MongoDB